internal/config/config.go

```diff
@@ -80,18 +90,127 @@ func LoadConfig() (*Config, error) {
 		ArkServer:        viper.GetString(ArkServer),
 		EsploraURL:       viper.GetString(EsploraURL),
 		CLNDatadir:       cleanAndExpandPath(viper.GetString(CLNDatadir)),
+		SentryDSN:        viper.GetString(SentryDSN),
+		SentryEnv:        viper.GetString(SentryEnv),
 		UnlockerType:     viper.GetString(UnlockerType),
 		UnlockerFilePath: viper.GetString(UnlockerFilePath),
 		UnlockerPassword: viper.GetString(UnlockerPassword),
 	}
 
+	if config.SentryDSN != "" {
+		if err := initSentry(config); err != nil {
+			return nil, fmt.Errorf("error initializing Sentry: %s", err)
+		}
+	}
+
 	if err := config.initUnlockerService(); err != nil {
 		return nil, err
 	}
 
 	return config, nil
 }
 
+func initSentry(config *Config) error {
+	err := sentry.Init(sentry.ClientOptions{
+		Dsn:              config.SentryDSN,
+		Environment:      config.SentryEnv,
+		AttachStacktrace: true,
+		ServerName:       GetHostname(),
+	})
+	if err != nil {
+		return err
+	}
+
+	setupLogrusHook()
+
+	return nil
+}
+
+func setupLogrusHook() {
+	levels := []log.Level{
+		log.PanicLevel,
+		log.FatalLevel,
+		log.ErrorLevel,
+		log.WarnLevel,
+	}
+
+	log.AddHook(&sentryHook{
+		levels: levels,
+	})
+}
+
+type sentryHook struct {
+	levels []log.Level
+}
+
+func (h *sentryHook) Levels() []log.Level {
+	return h.levels
+}
+
+func (h *sentryHook) Fire(entry *log.Entry) error {
+	if sentry.CurrentHub().Client() == nil {
+		return nil
+	}
+
+	// Skip if explicitly marked to skip Sentry
+	if skip, ok := entry.Data["skip_sentry"].(bool); ok && skip {
+		return nil
+	}
+
+	event := sentry.NewEvent()
+	event.Level = getSentryLevel(entry.Level)
+	event.Message = entry.Message
+	event.Extra = make(map[string]interface{})
+
+	for k, v := range entry.Data {
+		if k == "error" || k == "skip_sentry" {
+			continue
+		}
+		event.Extra[k] = v
+	}
+
+	if err, ok := entry.Data["error"].(error); ok {
+		event.Exception = []sentry.Exception{{
+			Value:      err.Error(),
+			Type:       fmt.Sprintf("%T", err),
+			Stacktrace: sentry.ExtractStacktrace(err),
+		}}
+	}
+
+	sentry.CaptureEvent(event)
+	return nil
+}
+
+func getSentryLevel(level log.Level) sentry.Level {
+	switch level {
+	case log.PanicLevel, log.FatalLevel:
+		return sentry.LevelFatal
+	case log.ErrorLevel:
+		return sentry.LevelError
+	case log.WarnLevel:
+		return sentry.LevelWarning
+	case log.InfoLevel:
+		return sentry.LevelInfo
+	case log.DebugLevel, log.TraceLevel:
+		return sentry.LevelDebug
+	default:
+		return sentry.LevelError
+	}
+}
+
+// GetHostname returns the hostname of the current machine
+func GetHostname() string {
```

Comment:
```suggestion
func getHostname() string {
```

and please move it to the bottom of the file